sw/device/silicon_creator/lib/drivers/pwrmgr.c

```diff
@@ -23,13 +24,27 @@ static_assert(
     PWRMGR_RESET_EN_EN_FIELD_WIDTH == 1,
     "RESET_EN field width changed, kAllResetsEnable may need to be updated");
 
+void pwrmgr_cdc_sync(uint32_t n) {
+  // We want to timeout if the CDC bit doesn't self clear.  We calculate the
+  // number of CPU cycles of 1 clock in the AON domain and add an additional
+  // 25 percent.
+  uint32_t cpu_cycle_timeout =
+      (uint32_t)kClockFreqCpuHz / (uint32_t)kClockFreqAonHz;
+  cpu_cycle_timeout += cpu_cycle_timeout / 4;
+
+  while (n--) {
+    ibex_mcycle_zero();
+    abs_mmio_write32(kBase + PWRMGR_CFG_CDC_SYNC_REG_OFFSET, kSyncConfig);
+    while (abs_mmio_read32(kBase + PWRMGR_CFG_CDC_SYNC_REG_OFFSET)) {
+      if (ibex_mcycle32() > cpu_cycle_timeout)
+        break;
+    }
+  }
```

Comment:
If there is a concern with an attacker bypassing synchronization, can you add something like this to try to detect this?

```c
if(launder32(abs_mmio_read32(kBase + PWRMGR_CFG_CDC_SYNC_REG_OFFSET))) {
  HARDENED_CHECK_GE(ibex_mcycle32() > cpu_cycle_timeout);
}
```